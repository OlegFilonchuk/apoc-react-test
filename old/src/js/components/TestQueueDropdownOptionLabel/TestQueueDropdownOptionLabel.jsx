import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './TestQueueDropdownOptionLabel.less';

export default function TestQueueDropdownOptionLabel(props) {
  const finalClassName = classNames('test-queue-dropdown-option-label', props.className);

  return (
    <span className={finalClassName}>
      <span className="sc-icon-solid-square" style={{ color: props.color }} />
      <span className="option-text">{props.text}</span>
    </span>
  );
}

TestQueueDropdownOptionLabel.propTypes = {
  /**
   * Text of option
   */
  text: PropTypes.string.isRequired,

  /**
   * Color of square's background
   */
  color: PropTypes.string.isRequired,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

TestQueueDropdownOptionLabel.defaultProps = {
  className: ''
};
