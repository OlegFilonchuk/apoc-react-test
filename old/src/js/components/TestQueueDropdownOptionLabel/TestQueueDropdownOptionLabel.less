.test-queue-dropdown-option-label {
  .sc-icon-solid-square {
    position: static;
    font-size: 12px;
    margin: 3px 6px 0 0;
  }
}

.sc-btn.sc-btn-dropdown {
  .test-queue-dropdown-option-label {
    .sc-icon-solid-square {
      font-size: 11px;
    }
  }
}
