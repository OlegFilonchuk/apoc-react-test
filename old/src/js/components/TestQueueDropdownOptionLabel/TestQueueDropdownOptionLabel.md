## Example
Purpose of this component is to be used together with [Dropdown](#dropdown).

    function callbackFn() {
      console.log('Dropdown state changed');
    }

     <Dropdown onChange={callbackFn} name={'test'} button={<DropdownButton/>}>
        <DropdownOption
          label={<TestQueueDropdownOptionLabel color={'red'} text={'Alpha'}/>}
          value={'alpha'}
        />
        <DropdownOption
          label={<TestQueueDropdownOptionLabel color={'blue'} text={'Beta'}/>}
          value={'beta'}
        />
        <DropdownOption
          label={<TestQueueDropdownOptionLabel color={'green'} text={'Charlie'}/>}
          value={'charlie'}
          isDisabled
        />
        <DropdownOption
          label={<TestQueueDropdownOptionLabel color={'yellow'} text={'Delta'}/>}
          value={'detla'}
          isSelected
        />
      </Dropdown>
