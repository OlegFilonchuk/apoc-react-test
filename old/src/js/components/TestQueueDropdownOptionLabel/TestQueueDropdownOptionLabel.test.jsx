import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TestQueueDropdownOptionLabel from './TestQueueDropdownOptionLabel';

describe('<TestQueueDropdownOption>', () => {
  it('renders normal', () => {
    const wrapper = shallow(<TestQueueDropdownOptionLabel color={'blue'} text={'some text'} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
