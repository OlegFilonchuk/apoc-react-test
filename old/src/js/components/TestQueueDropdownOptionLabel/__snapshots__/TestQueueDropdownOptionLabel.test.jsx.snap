// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestQueueDropdownOption> renders normal 1`] = `
<span
  className="test-queue-dropdown-option-label"
>
  <span
    className="sc-icon-solid-square"
    style={
      Object {
        "color": "blue",
      }
    }
  />
  <span
    className="option-text"
  >
    some text
  </span>
</span>
`;
